src/renderer/shared/api/xcm/lib/xcm-utils.ts

```diff
@@ -152,48 +155,50 @@ function getConcreteAssetLocation(assetLocation?: LocalMultiLocation) {
 }
 
 function getDestinationLocation(
-  originChain: Pick<Chain, 'parentId'>,
+  originChain: Pick<Chain, 'parentId' | 'specName'>,
   destinationParaId?: number,
   accountId?: AccountId,
 ) {
   if (originChain.parentId && destinationParaId) {
-    return getSiblingLocation(destinationParaId, accountId);
+    return getSiblingLocation(destinationParaId, accountId, originChain.specName as Network);
   }
 
   if (originChain.parentId) {
-    return getParentLocation(accountId);
+    return getParentLocation(accountId, originChain.specName as Network);
   }
 
   if (destinationParaId) {
-    return getChildLocation(destinationParaId, accountId);
+    return getChildLocation(destinationParaId, accountId, originChain.specName as Network);
   }
 
   return undefined;
 }
 
-function getAccountLocation(accountId?: AccountId) {
+function getAccountLocation(accountId?: AccountId, network?: string) {
```

Comment:
`network` seems too general coupled with `string` type. Have you tried something like this?
```javascript
function getAccountLocation<T extends Network>(accountId?: AccountId, network?: T) {
```